RecoEgamma/PhotonIdentification/interface/PhotonMVAEstimatorRun2Phys14NonTrig.h

```diff
@@ -62,29 +63,31 @@ class PhotonMVAEstimatorRun2Phys14NonTrig : public AnyMVAEstimatorRun2Base{
   ~PhotonMVAEstimatorRun2Phys14NonTrig();
 
   // Calculation of the MVA value
-  float mvaValue( const edm::Ptr<reco::Candidate>& particle);
+  float mvaValue(const edm::Ptr<reco::Candidate>& particle, const edm::Event&) const;
  
   // Utility functions
-  TMVA::Reader *createSingleReader(const int iCategory, const edm::FileInPath &weightFile);
+  std::unique_ptr<const GBRForest> createSingleReader(const int iCategory, const edm::FileInPath &weightFile) ;
 
-  inline int getNCategories(){return nCategories;};
-  bool isEndcapCategory( int category );
-  const inline std::string getName(){return name_;};
+  int getNCategories() const override { return nCategories; }
+  bool isEndcapCategory( int category ) const;
+  const std::string& getName() const override { return name_; }
 
   // Functions that should work on both pat and reco electrons
   // (use the fact that pat::Electron inherits from reco::GsfElectron)
-  void fillMVAVariables(const edm::Ptr<reco::Candidate>& particle);
-  int findCategory( const edm::Ptr<reco::Candidate>& particle);
+  std::vector<float> fillMVAVariables(const edm::Ptr<reco::Candidate>& particle, const edm::Event&) const override;
+  int findCategory(const edm::Ptr<reco::Candidate>& particle) const;
   // The function below ensures that the variables passed to MVA are 
   // within reasonable bounds
-  void constrainMVAVariables();
+  void constrainMVAVariables(AllVariables& vars) const;
 
   // Call this function once after the constructor to declare
   // the needed event content pieces to the framework
-  void setConsumes(edm::ConsumesCollector&&) override;
+  // DEPRECATED
+  void setConsumes(edm::ConsumesCollector&&) const override;
   // Call this function once per event to retrieve all needed
   // event content pices
-  void getEventContent(const edm::Event& iEvent) override;
+  // DEPRECATED
+  // void getEventContent(const edm::Event& iEvent) const override;
```

Comment:
deprecated and commented out means it's not needed anymore and can be removed.
What about other "DEPRECATED" that are still used??